src/main/java/org/apache/sling/jcr/contentloader/PathEntry.java

```diff
@@ -392,6 +414,28 @@ public boolean isIgnoredImportProvider(String extension) {
         return new HashSet<>(ignoreContentReaders);
     }
 
+    @Override
+    public boolean isImportProviderRequired(@NotNull String name) {
+        boolean required = false;
+
+        if (!this.requireContentReaders.isEmpty()) {
+            // a directive was supplied, so use a filter to check if the 
+            //  name ends with the suffix and is not listed in the ignored 
+            //  import provider set
+            required = this.requireContentReaders.stream()
+                            .anyMatch(ext ->
```

Comment:
imho this part of the code is unnecessary hard to read for that simple thing it does - probably move this extension extraction code to a separate method with proper name?

Reply:
See [e58b835](https://github.com/apache/sling-org-apache-sling-jcr-contentloader/pull/15/commits/e58b835b26e2acc4936d270c4a2ecc1ad19c575a)

Reply:
thanks, lgtm